### PR TITLE
feat: Introduce WebSocket helper for Node.js | `@hono/node-ws`

### PR DESCRIPTION
**Pull request by nakasyou**
(Recreated #492)
I created WebSocket helper implementation for Node.js

Usage:
```ts
import { createNodeWebSocket } from '@hono/node-ws'
import { Hono } from 'hono'
import { serve } from '@hono/node-server'

const app = new Hono()

const { injectServer, upgradeWebSocket } = createNodeWebSocket({ app })

app.get('/ws', upgradeWebSocket((c) => ({
  // https://hono.dev/helpers/websocket
})))

const server = serve(app)
injectServer(server)
```

### CODE REVIEW

**Comment by yusukebe** [comment]
@nakasyou 

Thanks! I'll review this later!

**Comment by nakasyou** [comment]
@yusukebe, I've converted to draft. Please wait a moment.

**Comment by nakasyou** [comment]
We can review now!

**Comment by yusukebe** [comment]
Hi @nakasyou 

CI is failing. Try to run `yarn install` on the top and push the new `yarn.lock`.

**Comment by nakasyou** [comment]
Hi @yusukebe 
> Try to run yarn install on the top and push the new yarn.lock.
I do it, could you re-run CI?

File: packages/node-ws/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+          socket.destroy()
+          return
+        }
+        res.wss.handleUpgrade(request, socket, head, (ws) => {
+          res.wss.emit('connection', ws, request)
+        })
+      })
+    },
+    upgradeWebSocket: (createEvents) =>
+      createMiddleware(async (c, next) => {
```
Comment (yusukebe) [inline packages/node-ws/src/index.ts:61]: It's better to make it a named function, see:

https://github.com/honojs/hono/pull/2615#discussion_r1591717786

File: packages/node-ws/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+## Usage
+
+```ts
+import { createNodeWebSocket } from '@hono/node-ws'
+import { Hono } from 'hono'
+import { serve } from '@hono/node-server'
+
+const app = new Hono()
+
+const { injectServer, upgradeWebSocket } = createNodeWebSocket({ app })
```
Comment (yusukebe) [inline packages/node-ws/README.md:14]: Should be the following?

```ts
const { injectWebSocket, upgradeWebSocket } = createNodeWebSocket({ app })
```

File: packages/node-ws/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+app.get('/ws', upgradeWebSocket((c) => ({
+  // https://hono.dev/helpers/websocket
+})))
+
+const server = serve(app)
+injectServer(server)
```
Comment (yusukebe) [inline packages/node-ws/README.md:21]: The same as above, this should be:

```ts
injectWebSocket
```

**Comment by yusukebe** [comment]
Hey @nakasyou 

I've updated it to fix some hassle issues and added the comments. Check them.

File: packages/node-ws/README.md (hunk 3 above)
Comment (nakasyou) [reply packages/node-ws/README.md]: I fixed at https://github.com/honojs/middleware/pull/503/commits/9e8bb486b94455be127b352f1a648f7574d0e10d.

File: packages/node-ws/src/index.ts (hunk 1 above)
Comment (nakasyou) [reply packages/node-ws/src/index.ts]: I fixed at https://github.com/honojs/middleware/pull/503/commits/36483096100b797ecc25d501cd77f633d702fe32.

**Comment by nakasyou** [comment]
@yusukebe, thank you for reviewing and working!
I fixed some issues.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@nakasyou 

Awesome! Let's land it!

**Pull request merged**